pkg/function/main.go

```diff
@@ -44,14 +45,39 @@ func main[T Inputs](fn SynthFunc[T], ir *InputReader, ow *OutputWriter) error {
 		if v.Field(i).IsNil() {
 			v.Field(i).Set(reflect.New(field.Type.Elem()))
 		}
-		err := ReadInput[client.Object](ir, tagValue, v.Field(i).Interface().(client.Object))
+		input := v.Field(i).Interface()
+
+		var obj client.Object
+		var custom bool
+		if o, ok := input.(client.Object); ok {
+			obj = o
```

Comment:
This is here to make it so the subsequent `ReadInput` call writes to `obj` right?

Reply:
Reading through this a couple of days after writing it took me an embarrassingly long time to figure out what's going on here lol. Pushed a commit to extract this logic out to a separate func.

In any case you're right. :)